### PR TITLE
Implemented #901,#902: new lite API

### PR DESCRIPTION
**Pull request by l-o-b-s-t-e-r**
Changed API for products search

### CODE REVIEW

**Comment by l-o-b-s-t-e-r** [comment]
#902

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java [hunk 1]
```diff
@@ -30,10 +30,10 @@
     String PRODUCT_API_COMMENT = "new android app";
 
 
-    @GET("api/v0/produit/{barcode}.json")
+    @GET("api/v0/product/{barcode}.json?fields=image_small_url,imageFrontUrl,image_ingredients_url,image_nutrition_url,url,code,traces_tags,ingredients_that_may_be_from_palm_oil_tags,additives_tags,allergens_hierarchy,manufacturing_places,nutriments,ingredients_from_palm_oil_tags,brands_tags,traces,categories,ingredients_text,product_name,generic_name,ingredients_from_or_that_may_be_from_palm_oil_n,serving_size,allergens,origins,stores,nutrition_grade_fr,nutrient_levels,countries,brands,packaging,labels,cities_tags,quantity,ingredients_from_palm_oil_n,image_url")
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: I was considering going even more radically minimal for quickmode, only displaying what's actually shown

```image_small_url,code,product_name,nutrition_grade_fr,brands,quantity```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java [hunk 2]
```diff
@@ -30,10 +30,10 @@
...
     Call<State> getProductByBarcode(@Path("barcode") String barcode);
 
-    @GET("cgi/search.pl?search_simple=1&json=1&action=process")
+    @GET("cgi/search.pl?search_simple=1&json=1&action=process&fields=image_small_url,imageFrontUrl,image_ingredients_url,image_nutrition_url,url,code,traces_tags,ingredients_that_may_be_from_palm_oil_tags,additives_tags,allergens_hierarchy,manufacturing_places,nutriments,ingredients_from_palm_oil_tags,brands_tags,traces,categories,ingredients_text,product_name,generic_name,ingredients_from_or_that_may_be_from_palm_oil_n,serving_size,allergens,origins,stores,nutrition_grade_fr,nutrient_levels,countries,brands,packaging,labels,cities_tags,quantity,ingredients_from_palm_oil_n,image_url")
     Call<Search> searchProductByName(@Query("search_terms") String name, @Query("page") int page);
```
Comment (teolemon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: Same for search (we don't display the nutriscore, but that would be nice to show)
image_small_url,code,product_name,nutrition_grade_fr,brands,quantity

**Comment by l-o-b-s-t-e-r** [comment]
@teolemon 
Changed API for search + added API call to get short product details for quick mode.
I didn't use RxJava for new API call, because (if I understood correctly from Slack) code refactoring will be made later and it makes no sense to use RxJava for only 1 call, so I tried to use the same codestyle as now.

**Comment by teolemon** [comment]
merge conflict to fix

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java [hunk 3]
```diff
@@ -30,10 +30,13 @@
     String PRODUCT_API_COMMENT = "new android app";
 
 
-    @GET("api/v0/produit/{barcode}.json")
-    Call<State> getProductByBarcode(@Path("barcode") String barcode);
+    @GET("api/v0/product/{barcode}.json?fields=image_small_url,imageFrontUrl,image_ingredients_url,image_nutrition_url,url,code,traces_tags,ingredients_that_may_be_from_palm_oil_tags,additives_tags,allergens_hierarchy,manufacturing_places,nutriments,ingredients_from_palm_oil_tags,brands_tags,traces,categories,ingredients_text,product_name,generic_name,ingredients_from_or_that_may_be_from_palm_oil_n,serving_size,allergens,origins,stores,nutrition_grade_fr,nutrient_levels,countries,brands,packaging,labels,cities_tags,quantity,ingredients_from_palm_oil_n,image_url")
```
Comment (policorn) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: The URL shall include 'links' field

Comment (l-o-b-s-t-e-r) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/network/OpenFoodAPIService.java]: Probably "link"
Sorry, some problems with merging...
Fixed

**Comment by l-o-b-s-t-e-r** [comment]
@policorn @itchix @teolemon 
ping ping))

**Comment by policorn** [comment]
Hi @l-o-b-s-t-e-r! Thanks for the fix. I've just tested it and found EMB Codes are not there (quite recent functionality introduced here: https://github.com/openfoodfacts/openfoodfacts-androidapp/commit/77e8587cfe115225912e06836b24f78193938ceb). Could you get it fixed? Thanks!
screenshot_1519691422

**Comment by l-o-b-s-t-e-r** [comment]
@policorn I've merged new changes...again)

**Comment by l-o-b-s-t-e-r** [comment]
@Karljoones 
Please, could you review this and merge, because I've already tired to fix merge conflicts (I'm doing this during last 3 days) :D

**Comment by Karljoones** [comment]
Hi @l-o-b-s-t-e-r, if i am 100% honest with you I am not fully up to date with the new API and would like to wait for @policorn  to have another look!

**Comment by teolemon** [comment]
The new API only sends what is requested. Thus if you don't request a specific field you won't get it and it might crash the app

**Comment by teolemon** [comment]
Otherwise no other changes

**Comment by policorn** [comment]
Hi @l-o-b-s-t-e-r, thanks for your patience. You're the first one dealing with Lite API and as you can see we have lots of PRs to be merged. I'm sure your implementation will set the base for future features implementations so merges will be less painful. Going to review this one right now J

**Comment by policorn** [comment]
Aaaaaaand, merged! Thanks @l-o-b-s-t-e-r!

**Pull request merged**